flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/CreditBasedPartitionRequestClientHandlerTest.java

```diff
@@ -419,6 +459,99 @@ public void testNotifyCreditAvailableAfterReleased() throws Exception {
 		}
 	}
 
+	@Test
+	public void testChannelReleasedBeforeDecodingBufferResponse() throws Exception {
+		testChannelReleasedOrRemovedBeforeDecodingBufferResponse(false);
+	}
+
+	@Test
+	public void testChannelRemovedBeforeDecodingBufferResponse() throws Exception {
+		testChannelReleasedOrRemovedBeforeDecodingBufferResponse(true);
+	}
+
+	@Test
+	public void testChannelReleasedBeforeReceivingBufferResponse() throws Exception {
+		testChannelReleasedOrRemovedBeforeReceivingBufferResponse(false);
+	}
+
+	@Test
+	public void testChannelRemovedBeforeReceivingBufferResponse() throws Exception {
+		testChannelReleasedOrRemovedBeforeReceivingBufferResponse(true);
+	}
+
+	private void testChannelReleasedOrRemovedBeforeDecodingBufferResponse(boolean isRemoved) throws Exception {
```

Comment:
`testChannelReleasedOrRemovedBeforeDecodingBufferResponse` seems not described properly. The real tests below is releasing channel after decoding buffer response, not before decoding.
`testDecodingBufferResponseBeforeReleasingChannel`?

Reply:
Renamed the methods.